### PR TITLE
Feature/propagate returnvalue

### PR DESCRIPTION
**Pull request by seesharper**
This PR adds support for propagating the return value from scripts.

We also align with CSI to return exit code 1 when a script throws or does not compile.

#158

### CODE REVIEW

File: src/Dotnet.Script.Tests/ScriptExecutionTests.cs [hunk 1]
```diff
@@ -75,6 +82,15 @@ public static void ShouldNotPassUnEscapedKnownArgumentToScript()
             Assert.DoesNotContain("-v", result.output);            
         }
 
+        [Fact]
+        public static void ShouldPropagateReturnValue()
+        {
+            var result = Execute($"{Path.Combine("ReturnValue", "ReturnValue.csx")}");
```
Comment (filipw) [inline src/Dotnet.Script.Tests/ScriptExecutionTests.cs]: since the errors are 1, let's use a different return value for this "happy path" test 😊

**Comment by filipw** [comment]
looks good!

**Comment by adamralph** [comment]
I'm not sure it's something you also want to align as well, but csi.exe does not let you return anything but an `int` from a script.

**Comment by filipw** [comment]
Yeah that's a good idea. Let's create the script explicitly as `int` here https://github.com/filipw/dotnet-script/pull/160/files#diff-5934cd0deb2fb20e20ed07e1c5eb9e23R155
Any other return type doesn't matter anyway as it's not going to be surfaced.

In CSI using a non-int return creates a compilation error.

File: src/Dotnet.Script.Tests/ScriptExecutionTests.cs (hunk 1 above)
Comment (seesharper) [reply src/Dotnet.Script.Tests/ScriptExecutionTests.cs]: I agree, lets use 42 :)

**Comment by seesharper** [comment]
Are we good now, @filipw ?

**Comment by filipw** [comment]
thanks!

**Pull request merged**